benchmark/url/url-searchparams-toString.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common.js');
+
+const values = {
+  noencode: {
+    foo: 'bar',
+    baz: 'quux',
+    xyzzy: 'thud',
+  },
+  encodemany: {
+    '\u0080\u0083\u0089': 'bar',
+    '\u008C\u008E\u0099': 'quux',
+    'xyzzy': '\u00A5q\u00A3r',
+  },
+  encodelast: {
+    foo: 'bar',
+    baz: 'quux',
+    xyzzy: 'thu\u00AC',
+  },
+  array: {
+    foo: [],
+    baz: ['bar'],
+    xyzzy: ['bar', 'quux', 'thud'],
+  },
+  multiprimitives: {
+    foo: false,
+    bar: -13.37,
+    baz: 'baz',
+  },
+};
+
+const bench = common.createBenchmark(main, {
+  type: ['noencode', 'encodemany', 'encodelast', 'array', 'multiprimitives'],
+  n: [1e6],
+});
+
+function main({ n, type }) {
+  const input = values[type];
+
+  // force optimisation
+  for (const name in values)
+    new URLSearchParams(values[name]).toString();
+
+  bench.start();
+  for (let i = 0; i < n; i++)
+    new URLSearchParams(input).toString();
```

Comment:
If you are only testing the performance of `toString,` it feels like initialization should not be inside the bench.

Reply:
Maybe we should spilt it into two, there is also no benchmark for the creation of the URLSearchParams object would that be good?